registry/registry.go

```diff
@@ -17,7 +19,12 @@ func NewRegistry(containerRepository ContainerRepository, serviceRepository Serv
 // Synchronize synchronizes registered services according to running containers
 func (r *Registry) Synchronize() {
 	registeredServicesIDs := r.serviceRepository.GetAllIds()
-	runningContainers := r.containerRepository.GetAll()
+	runningContainers, err := r.containerRepository.GetAll()
+
+	if err != nil {
+		log.Errorf("Error while fetching running containers: %+v", err)
```

Comment:
Just my opinion: I would return the error here and would write log on the top level. In this case it would be free of side effects. It also could make unit testing a bit easier.